astropy/table/table.py

```diff
@@ -882,10 +882,11 @@ def _convert_data_to_col(self, data, copy=True, default_name=None, dtype=None, n
         col : Column, MaskedColumn, mixin-column type
             Object that can be used as a column in self
         """
+        is_mixin = self._is_mixin_for_table(data)
+
         # Structured ndarray gets viewed as a mixin unless already a valid
         # mixin class
-        if (isinstance(data, np.ndarray) and len(data.dtype) > 1 and
-                not self._is_mixin_for_table(data)):
+        if isinstance(data, np.ndarray) and len(data.dtype) > 1 and not is_mixin:
```

Comment:
Not really for this PR, but I think for most `ndarray` subclasses one could just stick an `info` on and be done with it (can't do this with `ndarray` since it has `__slots__`, not `__dict__`). That might be better than the `.view` which destroys the properties of the subclass.

Reply:
I don't quite follow.  This is aimed explicitly at structured arrays, which have type of `ndarray`.

Reply:
It is just that the `isinstance` lets through any subclass, which is removed by the `.view`. But I think we can let this be covered by #8944.